aioredis/connection.py

```diff
@@ -405,6 +437,7 @@ async def read_response(self) -> Union[EncodableT, ResponseError, None]:
                 return None
             response = [(await self.read_response()) for _ in range(length)]
         if isinstance(response, bytes):
+            assert self.encoder is not None
```

Comment:
Why is this needed? If anything, we should probably raise RedisError instead.

Reply:
Not sure what the likelihood of this being `None` is. If you're confident it should never happen here, then an assert statement makes that clear (while telling mypy that the below code will never be run when it is `None`). If there's a chance that it might be `None` here, then we should raise `RedisError` instead.

Reply:
Hmm IIRC, we've had an issue with the buffer in the Encoder class before, but the Encoder object itself I've never seen become `None` before. So I think it's fine to remove the assertion.

Reply:
Relooking at it, I could make the check at the top of the function be `if not self._buffer or not self.encoder:`. That could also work, as it ensures encoder is not `None` and resolves the mypy error. The only code I see that sets it to `None` is the disconnect, so it's essentially the same check (although it's a public attribute, which suggests a user could assign `None` directly).